tests/sample_test.py

```diff
@@ -273,42 +271,52 @@ def test_is_comparator(self):
 
   def test_var_comparator(self):
     obj = SampleBase()
-    expect(obj.add_to_list).args( var('value1') )
-    expect(obj.add_to_list).args( var('value2') )
-    expect(obj.add_to_list).args( var('value3') ).at_least_once()
+    expect(obj.add_to_list).args(var('value1'))
+    expect(obj.add_to_list).args(var('value2'))
+    expect(obj.add_to_list).args(var('value3')).at_least_once()
 
     obj.add_to_list('v1')
     obj.add_to_list('v2')
     obj.add_to_list('v3')
     obj.add_to_list('v3')
-    self.assertRaises( UnexpectedCall, obj.add_to_list, 'v3a')
+    self.assertRaises(UnexpectedCall, obj.add_to_list, 'v3a')
 
-    assert_equals( 'v1', var('value1').value )
-    assert_equals( 'v2', var('value2').value )
-    assert_equals( 'v3', var('value3').value )
+    assert_equals('v1', var('value1').value)
+    assert_equals('v2', var('value2').value)
+    assert_equals('v3', var('value3').value)
 
   def test_spy(self):
     spy(SampleBase)
     obj = SampleBase()
-    assert_true(isinstance(obj,SampleBase))
+    assert_true(isinstance(obj, SampleBase))
 
     spy(obj.add_to_list)
     obj.add_to_list('v1')
     assert_equals(['v1'], list(obj._deque))
 
-    spy(obj.add_to_list).args( var('value2') )
+    spy(obj.add_to_list).args(var('value2'))
     obj.add_to_list('v2')
-    assert_equals(['v1','v2'], list(obj._deque))
-    assert_equals( 'v2', var('value2').value )
+    assert_equals(['v1', 'v2'], list(obj._deque))
+    assert_equals('v2', var('value2').value)
+
+    data = {'foo': 'bar'}
 
-    data = {'foo':'bar'}
-    def _fx(_data):
+    def _sfx(_data):
       _data['foo'] = 'bug'
 
-    spy(obj.add_to_list).side_effect(_fx, data)
+    spy(obj.add_to_list).side_effect(_sfx, data)
     obj.add_to_list('v3')
-    assert_equals(['v1','v2','v3'], list(obj._deque))
-    assert_equals({'foo':'bug'}, data)
+    assert_equals(['v1', 'v2', 'v3'], list(obj._deque))
+    assert_equals({'foo': 'bug'}, data)
+
+    capture = [0]
+
+    def _rfx(_deque):
```

Comment:
Ah, I see why you called it `return_effect`. While technically you could get away with this, I don't think that this is how we want to advertise the feature, as it's more for capturing the return value.